frontend/public/components/service-instance.tsx

```diff
@@ -0,0 +1,123 @@
+import { Cog, SectionHeading, navFactory, ResourceCog, ResourceLink, ResourceSummary, StatusWithIcon, Timestamp } from './utils';
+import { ResourceEventStream } from './events';
+import { Conditions } from './conditions';
+import { ServiceCatalogParameters, ServiceCatalogParametersSecrets } from './service-catalog-parameters';
+
+const ServiceInstancesReference: K8sResourceKindReference = 'ServiceInstance';
+
+const { common } = Cog.factory;
+
+const menuActions = [
+  ...common,
+];
+
+const ServiceInstanceDetails: React.SFC<ServiceInstanceDetailsProps> = ({obj: si}) => {
+  const siPlan = si.spec.clusterServicePlanExternalName || si.spec.servicePlanExternalName;
+  const siParameters = _.get(si, 'status.externalProperties.parameters', {});
+
+  return <React.Fragment>
+    <div className="co-m-pane__body">
+      <SectionHeading text="Service Instance Overview" />
+      <div className="row">
+        <div className="col-sm-6">
+          <ResourceSummary resource={si} showPodSelector={false} showNodeSelector={false} />
+        </div>
+        <div className="col-sm-6">
+          <dl className="co-m-pane__details">
+            <dt>External Name</dt>
+            {/* TODO:  add link */}
+            <dd>{si.spec.clusterServiceClassExternalName || si.spec.serviceClassExternalName}</dd>
+            <dt>Status</dt>
+            <dd><StatusWithIcon obj={si} /></dd>
+            <dt>Plan</dt>
+            <dd>{siPlan || '-'}</dd>
+          </dl>
+        </div>
+      </div>
+    </div>
+    <div className="co-m-pane__body">
+      <SectionHeading text="Conditions" />
+      <Conditions conditions={si.status.conditions} />
+    </div>
+    {!_.isEmpty(si.spec.parametersFrom) && <ServiceCatalogParametersSecrets obj={si} /> }
+    {!_.isEmpty(siParameters) && <ServiceCatalogParameters parameters={siParameters} /> }
+  </React.Fragment>;
+};
+
+const pages = [navFactory.details(ServiceInstanceDetails), navFactory.editYaml(), navFactory.events(ResourceEventStream)];
+export const ServiceInstanceDetailsPage: React.SFC<ServiceInstanceDetailsPageProps> = props =>
+  <DetailsPage
+    {...props}
+    kind={ServiceInstancesReference}
+    menuActions={menuActions}
+    pages={pages} />;
+ServiceInstanceDetailsPage.displayName = 'ServiceInstanceDetailsPage';
+
+const ServiceInstancesHeader = props => <ListHeader>
+  <ColHead {...props} className="col-md-3 col-sm-4 col-xs-6" sortField="metadata.name">Name</ColHead>
+  <ColHead {...props} className="col-md-3 col-sm-4 col-xs-6" sortField="metadata.namespace">Namespace</ColHead>
+  <ColHead {...props} className="col-md-2 col-sm-4 hidden-xs">Status</ColHead>
+  <ColHead {...props} className="col-md-2 hidden-sm hidden-xs" sortField="metadata.creationTimestamp">Created</ColHead>
+  <ColHead {...props} className="col-md-2 hidden-sm hidden-xs" sortField="spec.clusterServicePlanExternalName || spec.servicePlanExternalName">Plan</ColHead>
```

Comment:
`sortField="spec.clusterServicePlanExternalName || spec.servicePlanExternalName"`  Haha.  Nope.  That doesn't work.  Is there a workarouund?

Reply:
Pass `sortFunc` instead of `sortField`